polygeist/tools/cgeist/driver.cc

```diff
@@ -427,28 +425,34 @@ static LogicalResult optimize(mlir::MLIRContext &Ctx,
     // want to run optimizations, apart from the raising on the host code
     // itself, only on device code to avoid mismatches between the host code we
     // analyze and the actual host code generated by LLVM.
-    mlir::OpPassManager &OptPM =
-        (SYCLRaiseHost) ? PM.nest<gpu::GPUModuleOp>().nestAny() : PM.nestAny();
+    {
+      mlir::OpPassManager &OptPM = (SYCLRaiseHost)
+                                       ? PM.nest<gpu::GPUModuleOp>().nestAny()
+                                       : PM.nestAny();
+
+      OptPM.addPass(mlir::createCanonicalizerPass(CanonicalizerConfig, {}, {}));
+      OptPM.addPass(mlir::createCSEPass());
+      OptPM.addPass(polygeist::createMem2RegPass());
+      OptPM.addPass(mlir::createCanonicalizerPass(CanonicalizerConfig, {}, {}));
+      OptPM.addPass(mlir::createCSEPass());
+      OptPM.addPass(polygeist::createCanonicalizeForPass());
+      if (RaiseToAffine)
+        OptPM.addPass(polygeist::createRaiseSCFToAffinePass());
+      OptPM.addPass(polygeist::createReplaceAffineCFGPass());
+      OptPM.addPass(mlir::createCanonicalizerPass(CanonicalizerConfig, {}, {}));
+      OptPM.addPass(mlir::createCSEPass());
+      if (EnableLICM)
+        OptPM.addPass(polygeist::createLICMPass(
+            {options.getCgeistOpts().getRelaxedAliasing(), UseOpaquePointers}));
+      else
+        OptPM.addPass(mlir::createLoopInvariantCodeMotionPass());
+    }
 
-    OptPM.addPass(mlir::createCanonicalizerPass(CanonicalizerConfig, {}, {}));
-    OptPM.addPass(mlir::createCSEPass());
-    OptPM.addPass(polygeist::createMem2RegPass());
-    OptPM.addPass(mlir::createCanonicalizerPass(CanonicalizerConfig, {}, {}));
-    OptPM.addPass(mlir::createCSEPass());
-    OptPM.addPass(polygeist::createCanonicalizeForPass());
-    if (RaiseToAffine)
-      OptPM.addPass(polygeist::createRaiseSCFToAffinePass());
-    OptPM.addPass(polygeist::createReplaceAffineCFGPass());
-    OptPM.addPass(mlir::createCanonicalizerPass(CanonicalizerConfig, {}, {}));
-    OptPM.addPass(mlir::createCSEPass());
-    if (EnableLICM)
-      OptPM.addPass(polygeist::createLICMPass(
-          {options.getCgeistOpts().getRelaxedAliasing(), UseOpaquePointers}));
-    else
-      OptPM.addPass(mlir::createLoopInvariantCodeMotionPass());
     if (EnableLoopInternalization)
-      OptPM.addPass(polygeist::createLoopInternalizationPass(
+      PM.addPass(polygeist::createLoopInternalizationPass(
           {options.getCgeistOpts().getRelaxedAliasing()}));
+    mlir::OpPassManager &OptPM =
```

Comment:
To improve visibility, can we put line 454 to 473 under braces too? And put `createInlinePass` after.

Reply:
Yes, will do.